insights/parsers/messages.py

```diff
@@ -86,3 +86,23 @@ def get(self, s):
                     msg_info['procname'] = info_splits[4]
                 r.append(msg_info)
         return r
+
+    def get_after(self, timestamp, lines=None):
+        """
+        Get a list of lines after the given time stamp.
+
+        Parameters:
+            timestamp(datetime.datetime): log lines after this time are
+                returned.
+            lines(list): the list of log lines to search (e.g. from a get).
+                If not supplied, all available lines are searched.
+
+        Returns:
+            (list): The list of log lines with time stamps after the given
+                date and time.
+
+        *NOTE*: Because /var/log/messages timestamps by default have no year,
+        you must supply the year in your timestamp as 1900.  This may also
+        cause confusion around the end of the year.
```

Comment:
Yeah - some log files, like Apache's access log, get the date stamp from when the request started but are written to the log when the request finishes; long requests can make the log look like it went backward in time.  The other problem with just checking an end-of-year rollover is that a search for dates in January 2017 will give logs written in December the year of 2017 - which makes them appear as if they're in the log range when they are most likely to be from 2016.  Because this method is a generator we can't go back and retroactively re-parse the logs now that we know that they probably started in December 2016.